git.md

```diff
@@ -1,2 +1,127 @@
+Testing your code is important, particularly when sharing it with
+others. Automated tests are the best but not the only way to test the effects
+of the changes you made. However you did it, explain how you ensured that the
+code behaves as you expect, and how someone else may reproduce your results.
+
+## 6. Structure each commit message like an email
+
+Like an email, commit messages have a subject and an optional body. The subject
+should start with a line summarizing your changes (aim for 50 characters or
+less). Use the imperative, present tense -- for example, "Fix typo",
+not "Fixed typo" or "Fixes typo". The message body, if present,
+should follow a blank line; wrap around 70 characters; and explain the
+motivation for the change ("why", not "what" or "how").
+
+See also:
+
+* [Example of a commit message](https://github.com/2DegreesInvesting/resources/issues/74).
+* <https://r-pkgs.org/git.html#commit-best-practices.>
+
+## 7. Agree on a workflow
+
+Choose a workflow based on your team's needs and preferences. However you
+choose to work, just make sure to agree on a common workflow that everyone
+follows:
+
+GitHub workflow (a.k.a Git feature branch workflow:
+
+* Uses a central repository. 
+* Uses a single long-lived branch: "main"; it never contains broken code.
+* Developers work on "feature branches, share them via pull requests, and
+peers review the changes before they become a part of the main codebase.
+* A new release is created by tagging a particular commit on the main branch.
+
+Gitflow workflow:
+
+* Like the GitHub workflow but uses two long-lived branches: "main"
+and "develop". The main branch stores the official release history,
+and the develop branch serves as an integration branch for features.
+
+See also: <https://www.atlassian.com/git/tutorials/comparing-workflows>.
+
+## 8. Use tools to be more productive
+
+Choose the tools that help you work more comfortably, more productively,
+and with fewer errors.
+
+See also:
+
+* [Install a Git client](https://happygitwithr.com/git-client.html)
+* [oh-my-zsh: Unleash your terminal like never before](https://ohmyz.sh/).
+* [gh: Take GitHub to the command line](https://cli.github.com/).
+* [Git from RStudio](https://rstudio.com/resources/webinars/managing-part-2-github-and-rstudio/).
+
+## References
+
+[Version control best practices](https://www.git-tower.com/blog/version-control-best-practices/)
+
+## How
+
+* [Git practices by example](TODO)
+* <https://github.com/2DegreesInvesting/coding-helpdesk#coding-helpdesk>.
+* <https://github.com/git-guides/>.
+* <https://coderefinery.github.io/git-intro/>.
+* <https://bit.ly/book-git-in-practice (Part 4: Git best practices)>.  
+* <https://dev.to/lydiahallie/cs-visualized-useful-git-commands-37p1>.
+
+## Discuss
+
+* Jackson: Always include files like LICENSE, README.md, and consider
+.gitignore and CONTRIBUTING.md.  
+* Taylor: Develop a consensus for types of changes, e.g. "amend" (something
+I thought I had already done).
```

Comment:
@tposey28, Can you expand a bit on this idea, maybe with a few examples? Also which subsection would you fit it, maybe workflow?

Reply:
I think it goes in workflow.
My intent with this was that a developer who wants to amend a commit can do one of two things:
1. They can literally amend their latest commit, using the `-amend` command. It can be used to change the contents or the commit message, but is best only used if the commit hasn't been pushed yet (if it has then a merge conflict will occur, which is a bit confusing, plus at this point the commit has been "shared" in a sense).
2. If the commit is no longer the latest (or we want to avoid using the `-amend` flag because it's technically rewriting history) then I was suggesting that one could literally use a commit message like the following: "Amend: Added behavior to script". This isn't something I've ever heard of in practice, it's just an idea I had that could make following the flow easier in a PR.

Best case scenario you never need to amend ;)

Reply:
https://careerkarma.com/blog/git-commit-amend/

Reply:
This is an interesting idea, but it seems a bit niche/ not frequently used to be included as a "best practice". Maybe we could add an addendum with "nice-to-haves" or something like that? 
@maurolepore @MonikaFu @cjyetman @everyone thoughts?

Reply:
@jdhoffa I like the idea of nice-to-have's and what @tposey28 mentions sounds like an example of a good commit message also so maybe we could use it like that as well.

Reply:
How about writing something like this?:

```
## Agree on a workflow

GitHub workflow

... bla bla

Git workflow

... bla bla

A team may agree on custom conversions beyond the "vanilla" workflows listed above. 
For example, you may agree to prefix a commit title with "WIP", "FIXME", "Amend: " to
communicate with your team succinctly via a vocabulary you all share.
```

Reply:
@maurolepore sounds great to me